pkg/rke2/kp.go

```diff
@@ -0,0 +1,52 @@
+package rke2
+
+import (
+	"context"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const kubeProxyChart = "rke2-kube-proxy"
+
+// setKubeProxyDisabled determines if a cluster already has kube proxy deployed as a chart, if so
+// disables running kubeproxy as a static pod.
+func setKubeProxyDisabled(clx *cli.Context, cfg *cmds.Server) func(context.Context, <-chan struct{}, string) error {
+	return func(ctx context.Context, apiServerReady <-chan struct{}, kubeConfigAdmin string) error {
+		go func() {
+			<-apiServerReady
+			logrus.Info("Checking for kube proxy as a chart")
+
+			restConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigAdmin)
+			if err != nil {
+				logrus.Fatalf("kp: new k8s cient: %s", err.Error())
```

Comment:
this is probably my own typo
```suggestion
				logrus.Fatalf("kp: new k8s client: %s", err.Error())
```